File: herdr-picker-scripts/herdr-picker-focus-agent.py
#!/usr/bin/env python3
"""Focus the agent pane selected in herdr-picker."""

import json
import subprocess
import sys

context = json.load(sys.stdin)
selection = context["selections"][context["step"]]["value"]
subprocess.run(["herdr", "agent", "focus", selection["pane_id"]], check=True)

File: herdr-picker-agents/src/bin/herdr-picker-focus-agent.rs
//! Submit command: focuses the agent pane selected in herdr-picker.

use std::process::ExitCode;

use herdr_picker_sdk::{run, submit};

fn main() -> ExitCode {
    let result = submit("agent.focus", "pane_id", "target");
    run(env!("CARGO_BIN_NAME"), result)
}
